### PR TITLE
txscript: Rework check signature encoding test.

### PR DESCRIPTION
**Pull request by davecgh**
This reworks the tests for `CheckSignatureEncoding` to test for the explicit reason for the failure to ensure that each test is actually
testing the intended condition and be more consistent with the more recent tests.

It also adds a few more tests for some corner cases where signatures will ultimately fail validation due to ECDSA requirements, but are still valid encodings from the standpoint of preventing malleability.

### CODE REVIEW

File: txscript/consensus_test.go [hunk 1]
```diff
@@ -74,172 +75,205 @@ func TestExtractCoinbaseNullData(t *testing.T) {
...
+		// Although signatures with R == 0 will ultimately be invalid, it is
+		// considered a valid encoding from the standpoint of preventing
+		// malleability.
+		name: "R == 0",
+		sig: hexToBytes("30250201000220181522ec8eca07de4860a4acdd12909d831cc5" +
+			"6cbbac4622082221a8768d1d09"),
+		err: nil,
+	}, {
+		// Although signatures with R == N will ultimately be invalid, it is
+		// considered a valid encoding from from the standpoint of preventing
```
Comment (dnldd) [inline txscript/consensus_test.go:213]: repeated `from`.

Comment (davecgh) [reply txscript/consensus_test.go]: Nice catch.  Fixed.
